homeassistant/components/nest/sensor_sdm.py

```diff
@@ -89,28 +96,17 @@ def device_model(self):
         # The API intentionally returns minimal information about specific
         # devices, instead relying on traits, but we can infer a generic model
         # name based on the type
-        if self._device.type == "sdm.devices.types.CAMERA":
-            return "Camera"
-        if self._device.type == "sdm.devices.types.DISPLAY":
-            return "Display"
-        if self._device.type == "sdm.devices.types.DOORBELL":
-            return "Doorbell"
-        if self._device.type == "sdm.devices.types.THERMOSTAT":
-            return "Thermostat"
-        return None
+        return DEVICE_TYPE_MAP.get(self._device.type, None)
 
     async def async_added_to_hass(self):
         """Run when entity is added to register update signal handler."""
-
-        async def async_update_state():
-            """Update sensor state."""
-            await self.async_update_ha_state(True)
-
         # Event messages trigger the SIGNAL_NEST_UPDATE, which is intercepted
         # here to re-fresh the signals from _device.  Unregister this callback
         # when the entity is removed.
         self.async_on_remove(
-            async_dispatcher_connect(self.hass, SIGNAL_NEST_UPDATE, async_update_state)
+            async_dispatcher_connect(
+                self.hass, SIGNAL_NEST_UPDATE, self.async_update_ha_state
```

Comment:
I think we can replace `self.async_update_ha_state` with `self.async_write_ha_state`.

Reply:
When I replace that, the tests fail because the update is not reflected in state.   I think the only difference i see is the "async" keyword prefix.  I'm fairly new to python async so i'm missing the answer

Reply:
It should work. Maybe something else in the tests isn't set up correctly?

Reply:
fixed it in a404a220b3

Reply:
(and then I had a whoopsie and reverted the change for legacy in 6b44456)

Reply:
> In the fake subscriber in the tests we have decorated a coroutine function `async_device_manager` as a property. I dont think that is correct.

Thanks, I'll fix this in the underlying library, then update HA and tests.